sklearn/tree/_criterion.pxd

```diff
@@ -40,6 +40,14 @@ cdef class Criterion:
     cdef double weighted_n_left          # Weighted number of samples in the left node
     cdef double weighted_n_right         # Weighted number of samples in the right node
 
+    cdef double* node_sum                # An array of the current node value
+    cdef double* node_sum_left           # An array of the left node value
+    cdef double* node_sum_right          # An array of the right node value
+
+    cdef double yw_sq_sum                # Cumulative square in the current node
+    cdef double yw_sq_sum_left           # Cumulative square in the left node
+    cdef double yw_sq_sum_right          # Cumulative square in the right node                
```

Comment:
With gini you need to square the count of your labels, so it's not as simple to keep a running sum. If you were to change this value after moving one label over, you'd need to subtract 2n-1 or add 2n+1 where n is your previous observed label count. I attempted this optimization when I did the gini proxy impurity improvement, and it didn't seem to have any speed gain. It was a disappointment.